src/main/java/org/rulez/demokracia/pdengine/MessageSigner.java

```diff
@@ -0,0 +1,102 @@
+package org.rulez.demokracia.pdengine;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+
+public class MessageSigner {
+
+	private PublicKey pubkey;
+	private PrivateKey privkey;
+
+    private static class Storage {
```

Comment:
Why do we need this class?

Reply:
Using static initialization is a thread-safe singleton implementation.

Reply:
Refer to the fields directly from within the class, make them private, make all methods non-static, and implement getInstance() as the only static method.